include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
@behlendorf ....today even though we use DTRACE_PROBE2 and pass two arguments db and mls, we actually are using only the db arg to log various information in the trace. mls arg is ignored, so, question is do we want to still use DTRACE_PROBE1 passing mls as the only arg and if so, then do we want to log NULL in output or extract sth useful from the multilist arg